File: Java Data/T 01 Patterns/Pattern_02_Right_Angled_Triangle_Pattern.java
import java.util.Scanner;

// *
// * *
// * * *
// * * * *
// * * * * *

public class Pattern_02_Right_Angled_Triangle_Pattern {
    public static void printRightAngledTrianglePattern(int n) {
        // Outer loop controls the rows. It runs 'n' times (once for each row)
        for (int i = 1; i <= n; i++) {
            // Inner loop for columns, iterating up to the current row number 'i'
            for (int j = 1; j <= i; j++) {

                System.out.print("* ");
            }
            // Move to the next line after each row is complete
            System.out.println();
        }
    }

    public static void main(String[] args) {

        Scanner scanner = new Scanner(System.in);

        System.out.print("Enter the number of rows for the right-angled triangle pattern: ");
        int n = scanner.nextInt();

        printRightAngledTrianglePattern(n);

        scanner.close();
    }
}

File: 00-Patterns-Python/Pattern_02_Right_Angled_Triangle_Pattern.py
# *
# * *
# * * *
# * * * *
# * * * * *

def print_right_angle_triangle_pattern(n):
    # Outer loop controls the rows. It runs 'n' times (once for each row)
    for i in range(1, n + 1, 1):

        # Inner loop for columns, iterating up to the current row number 'i
        for j in range(1, i + 1, 1):
            print("*", end=" ")

        # Move to the next line after each row is complete
        print()

num = input("Enter the number of rows for the right-angled triangle pattern: ")
print_right_angle_triangle_pattern(int(num))
